### PR TITLE
[BEAM-5419] Add Flink multi-version builds

### PR DESCRIPTION
**Pull request by mxm**
The PR contains three commits in this order:

## [BEAM-5419] Add Flink multi-version build layout

This enables to build against multiple Flink versions. Each version lives in a
separate project folder and loads the main 'flink_runner.gradle' script. Each
project can configure a custom Flink version and a custom source folder
location. This will allow to do overrides in the future when the source can not
be made compatible across different Flink versions.

## [BEAM-5267] Add Flink v1.6 target for FlinkRunner 

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by mxm** [comment]
Retest this please

**Comment by mxm** [comment]
CC @angoenka @robertwb @tweise @jsteggink

**Comment by mxm** [comment]
Run Java PreCommit

**Comment by mxm** [comment]
I decided to remove the 1.7 target for now because I saw test failures in PreCommit (only on Jenkins, not local). We should add 1.7 in a separate PR.

**Comment by mxm** [comment]
Run Website PreCommit

**Comment by mxm** [comment]
Flaky test in `fn-execution`: https://builds.apache.org/job/beam_PreCommit_Java_Commit/3007/

**Comment by mxm** [comment]
Run Java PreCommit

**Comment by angoenka** [comment]
Thanks Max.
LGTM.

File: settings.gradle [hunk 1]
```diff
@@ -38,12 +38,20 @@ include "beam-runners-direct-java"
...
+/* Begin Flink Runner related settings */
+// 1.5 version with Scala 2.11 suffix
 include "beam-runners-flink_2.11"
 project(":beam-runners-flink_2.11").dir = file("runners/flink")
 include "beam-runners-flink_2.11-job-server"
 project(":beam-runners-flink_2.11-job-server").dir = file("runners/flink/job-server")
 include "beam-runners-flink_2.11-job-server-container"
 project(":beam-runners-flink_2.11-job-server-container").dir = file("runners/flink/job-server-container")
+// 1.6 version
+include "beam-runners-flink-1.6"
```
Comment (tweise) [inline settings.gradle]: Thanks for not including a scala suffix!

**Comment by mxm** [comment]
Run Website PreCommit

**Pull request merged**